### ISSUE TITLE
Data-Printer: tests fail on Perl 5 blead; may not install on perl-5.26.0

### ISSUE THREAD

**Issue by jkeenan** [garu/Data-Printer#111]
Today I attempted to assess the compatibility of Data-Printer with the upcoming perl-5.26.0.  I build perl5 blead and installed the latest version of App::cpanminus against that perl.  I then tried to use 'cpanm' to install Data::Printer against that perl.  I got these results:
```
$ cat /home/jkeenan/.cpanm/work/1491147343.21714/build.log
cpanm (App::cpanminus) 1.7042 on perl 5.026000 built for x86_64-linux
Work directory is /home/jkeenan/.cpanm/work/1491147343.21714
You have make /usr/bin/make
You have LWP 6.24
You have /bin/tar: tar (GNU tar) 1.28
Copyright (C) 2014 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by John Gilmore and Jay Fenlason.
You have /usr/bin/unzip
Searching Data::Printer () on cpanmetadb ...
--> Working on Data::Printer
Fetching http://www.cpan.org/authors/id/G/GA/GARU/Data-Printer-0.38.tar.gz
-> OK
Unpacking Data-Printer-0.38.tar.gz
Entering Data-Printer-0.38
Checking configure dependencies from META.json
Checking if you have ExtUtils::MakeMaker 6.58 ... Yes (7.24)
Configuring Data-Printer-0.38
Running Makefile.PL
Checking if your kit is complete...
Looks good
Generating a Unix-style Makefile
Writing Makefile for Data::Printer
Writing MYMETA.yml and MYMETA.json
-> OK
Checking dependencies from MYMETA.json ...
Checking if you have Sort::Naturally 0 ... Yes (1.03)
Checking if you have Term::ANSIColor 3 ... Yes (4.06)
Checking if you have File::Temp 0 ... Yes (0.2304)
Checking if you have Clone::PP 0 ... Yes (1.06)
Checking if you have ExtUtils::MakeMaker 0 ... Yes (7.24)
Checking if you have Test::More 0.88 ... Yes (1.302078)
Checking if you have File::Spec 0 ... Yes (3.67)
Checking if you have Carp 0 ... Yes (1.42)
Checking if you have File::HomeDir 0.91 ... Yes (1.00)
Checking if you have Fcntl 0 ... Yes (1.13)
Checking if you have Scalar::Util 0 ... Yes (1.47)
Checking if you have Package::Stash 0.3 ... Yes (0.37)
Checking if you have version 0.77 ... Yes (0.9917)
Building and testing Data-Printer-0.38
cp lib/Data/Printer.pm blib/lib/Data/Printer.pm
cp lib/Data/Printer/Filter/Digest.pm blib/lib/Data/Printer/Filter/Digest.pm
cp lib/Data/Printer/Filter.pm blib/lib/Data/Printer/Filter.pm
cp lib/Data/Printer/Filter/DB.pm blib/lib/Data/Printer/Filter/DB.pm
cp lib/Data/Printer/Filter/DateTime.pm blib/lib/Data/Printer/Filter/DateTime.pm
cp lib/DDP.pm blib/lib/DDP.pm
PERL_DL_NONLAZY=1 "/home/jkeenan/testing/blead/bin/perl" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t
# Beginning Data::Printer tests in linux with Perl 5.026000, /home/jkeenan/testing/blead/bin/perl
# Testing Data::Printer 0.38
t/00-load.t ................. ok
t/00.2-begin_load.t ......... ok
t/01-p.t .................... ok
t/02-colors.t ............... ok
t/02.2-autocolor.t .......... skipped: IO::Pty::Easy required for auto-colored tests
t/03-conf.t ................. ok
t/03-conf_not_ref.t ......... ok
t/04-multiline.t ............ ok
t/04-multiline2.t ........... ok
t/05-obj.t .................. ok
t/06-obj2.t ................. ok
t/07-sort.t ................. ok
t/08-deparse.t .............. ok
t/09-alias.t ................ ok
t/10-filter.t ............... ok
t/11-aliased_with_filter.t .. ok
t/12-filter_class.t ......... ok
t/13-filter_datetime.t ...... ok
DBIx::Class::Storage::DBI::_warn_undetermined_driver(): This version of DBIC does not yet seem to supply a driver for your particular RDBMS and/or connection method ('DBM'). While we will attempt to continue anyway, the results are likely to be underwhelming. Please upgrade DBIC, and if this message does not go away, file a bugreport including the following info:
{
  DBD => "DBD::DBM",
  DBD_VER => "0.08",
  DBIC_DRIVER => "DBIx::Class::Storage::DBI",
  DBIC_DSN => "dbi:DBM(RaiseError=1):",
  DBIC_VER => "0.082840",
  DBI_VER => "1.636"
}
t/13.2-filter_db.t .......... ok
t/13.3-filter_digest.t ...... ok
t/14-local_conf.t ........... ok
t/15-rc_file.t .............. ok
t/16-rc_file2.t ............. ok
t/16.2-rc_overwrite.t ....... ok
t/16.3-rc_env.t ............. ok
t/16.4-rc_env2.t ............ ok
t/16.5-rc_env3.t ............ ok
t/17-parallel.t ............. ok
t/18-class_method.t ......... ok
t/19-tied.t ................. ok
t/20-handles.t .............. ok
t/21-ddp.t .................. ok
t/22-class_method.t ......... ok
t/23-caller_info.t .......... ok
t/24-no_prototypes.t ........ ok
t/25-weak.t ................. ok
t/26-tainted.t .............. ok
*** WARNING *** rc file '/tmp/kR8rdOSVCe/my_home/.dataprinter' must NOT be writeable to other users. Skipping.
t/26.2-tainted_rc.t ......... ok
t/26.7-unicode.t ............ ok

#   Failed test 'pass-through return (hash scalar)'
#   at t/27-pass_through.t line 47.
#                   '1'
#     doesn't match '(?^:^1/\d+$)'
# Looks like you failed 1 test of 24.
t/27-pass_through.t ......... 
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/24 subtests 

#   Failed test 'pass-through return (hash scalar)'
#   at t/27.2-pass_through.t line 47.
#                   '1'
#     doesn't match '(?^:^1/\d+$)'
# Looks like you failed 1 test of 24.
t/27.2-pass_through.t ....... 
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/24 subtests 

#   Failed test 'pass-through return (hash scalar)'
#   at t/27.3-pass_through-DDP.t line 47.
#                   '1'
#     doesn't match '(?^:^1/\d+$)'
# Looks like you failed 1 test of 24.
t/27.3-pass_through-DDP.t ... 
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/24 subtests 

#   Failed test 'pass-through return (hash scalar)'
#   at t/27.4-pass_through-DDP.t line 47.
#                   '1'
#     doesn't match '(?^:^1/\d+$)'
# Looks like you failed 1 test of 24.
t/27.4-pass_through-DDP.t ... 
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/24 subtests 
t/28-void_return.t .......... ok
t/29-output.t ............... ok
t/30-print_escapes.t ........ ok
t/31-bad_parameters.t ....... ok
t/32-quote_keys.t ........... ok
t/33-end_separator.t ........ ok
t/33-separator.t ............ ok
t/34-show_readonly.t ........ ok
t/35-vstrings.t ............. ok
t/36-valign.t ............... ok
t/37-format.t ............... ok
t/38-lvalue.t ............... ok
t/39-seen_override.t ........ ok
t/40-escape_chars.t ......... ok
t/41-yaml.t ................. ok
t/42-alignment.t ............ ok
t/43-_is_number.t ........... ok
t/44-number-colors.t ........ ok
t/45-np.t ................... ok

Test Summary Report
-------------------
t/27-pass_through.t       (Wstat: 256 Tests: 24 Failed: 1)
  Failed test:  6
  Non-zero exit status: 1
t/27.2-pass_through.t     (Wstat: 256 Tests: 24 Failed: 1)
  Failed test:  6
  Non-zero exit status: 1
t/27.3-pass_through-DDP.t (Wstat: 256 Tests: 24 Failed: 1)
  Failed test:  6
  Non-zero exit status: 1
t/27.4-pass_through-DDP.t (Wstat: 256 Tests: 24 Failed: 1)
  Failed test:  6
  Non-zero exit status: 1
Files=62, Tests=454,  4 wallclock secs ( 0.16 usr  0.04 sys +  3.80 cusr  0.27 csys =  4.27 CPU)
Result: FAIL
Failed 4/62 test programs. 4/454 subtests failed.
Makefile:885: recipe for target 'test_dynamic' failed
make: *** [test_dynamic] Error 255
-> FAIL Installing Data::Printer failed. See /home/jkeenan/.cpanm/work/1491147343.21714/build.log for details. Retry with --force to force install it.
```
If we are getting these failures at this point in the annual perl 5 release cycle, then there is a high likelihood that Data-Printer will not test and install on perl 5.26.0.  Could you investigate?

Thank you very much.
Jim Keenan

**Comment by jkeenan** [comment]
perl-5.24.1 with latest CPAN release of Data-Printer, installed via ```cpanm```:
```
$ perl -v | head -2 | tail -1
This is perl 5, version 24, subversion 1 (v5.24.1) built for x86_64-linux

[perl] 534 $ perl -MData::Printer -lwe 'my %h = (alpha => q|beta|, gamma => q|delta|); my $rv = p %h; print $rv'{
    alpha   "beta",
    gamma   "delta"
}

2/8
```

perl-5 blead with latest CPAN release of Data-Printer, installed via ```cpanm```:
```
[blead] 548 $ ./bin/perl -v | head -2 | tail -1
This is perl 5, version 26, subversion 0 (v5.26.0 (v5.25.11-27-g1b92e69)) built for x86_64-linux

$ ./bin/perl -Ilib -I/home/jkeenan/.cpanm/work/1491147343.21714/Data-Printer-0.38/blib/lib -MData::Printer -lwe 'my %h = (alpha => q|beta|, gamma => q|delta|); my $rv = p %h;p $rv'
{
    alpha   "beta",
    gamma   "delta"
}

2
```

**Comment by garu** [comment]
Hi @jkeenan! Thanks for the bump :)

Looks like #97 does fix everything, so I've went ahead and released Data::Printer 0.39 to CPAN.

Cheers!
